### DISCUSSION TITLE
Harberger Tax for Livestreaming NFT

### DISCUSSION THREAD

**Discussion by chrishobcroft** [NFTsTv/Broadcastr#57] [Ideas]
## Introduction

Livestreaming NFTs are an emerging asset class for AV (audio-visual) media.

One unanswered question around LNFTs is how the economics will work for those who create LNFTs, and what the incentive is for them to continue to publish content to that LNFT once it has been sold.

Use of Harberger Tax presents an opportunity to develop "Patreon-like" funding models, as a way to create engagement between Publisher and Consumer.

This post seeks to outline a concept, and provoke a discussion about experimenting with such mechanisms.

### What are Livestreaming NFTs

Livestreaming NFTs (LNFTs) are NFTs which display linearly-broadcast livestreaming video content. See below for a screenshot showing a LNFT being viewed in NFT explorer app [NFTScan](https://nftscan.com):

![Screenshot_20230217-191611_DuckDuckGo](https://user-images.githubusercontent.com/2212651/220350582-9de4a962-1a28-4cd1-b533-d5a34d4a6af5.png)

### What is Harberger Tax?

This post will assume a basic working knowledge of Harberger Tax, but in case you don't have that, here are some articles which might help:

https://en.m.wikipedia.org/wiki/Harberger_Tax
https://thisartworkisalwaysonsale.com/
https://outland.art/simon-de-la-rouviere-this-artwork-is-always-on-sale/
https://mobile.twitter.com/VitalikButerin/status/1555685981592104961
https://blog.wildcards.world/build-your-own-harberger-tax-app-part-1/

## Problem statement

LNFTs created by Publishers of livestreaming content, and sold to Consumers of livestreaming content, lack incentives for Publishers to continue to publish after the sale of the LNFT.

Consumers who have paid once to purchase may therefore feel dissatisfied that their LNFTs appear somewhat "dead".

Publishers who have been paid once when selling the LNFT, may seek continuing revenue streams elsewhere, in order to fund their living costs.

These problems undermine the economic viability of LNFTs.

## LNFT with Harberger Tax

This section seeks to outline a "straw man" mechanism for how to implement Harberger Tax for LNFTs.

### Actors

Actors in this system are:

- *Publisher*: who mints the LNFT collection, and controls the keys to publish content to the LNFT.
- *Consumer*: who holds the LNFT in their wallet.

### Potential operation

When minting the collection, the Publisher provides name/description, and defines:

i) a mint price, at which any Consumer can mint their own LNFT,
ii) the tax rate, a % of the Consumer's defined sale price to be paid annually to the Publisher,
iii) the total supply of LNFTs that can exist.

A Consumer pays the mint price plus gas, to mint their own LNFT. At this point, a Consumer must also:
a) set their sale price, which another Consumer can pay if they want to force a sale. This can default to the mint price to reduce friction.
b) provide funds to automatically pay the % tax to the Publisher. This can be avoided if the Consumer already has non-zero funds deposited.

At this point, any other Consumer can pay the sale price and force a sale, and take ownership of the LNFT. At this point they must also set their sale price and provide funds to pay the tax.

If the funds to pay the % tax run out, then the LNFT can be "repossessed" by the Publisher for zero (plus fees).

The Publisher can, at any time, change:
a) the mint price for minting a fresh LNFT (if stock available).
b) the annual % rate for tax,
c) the total number of LNFTs (with a lower bound of minted tokens in circulation).

## Example Consumer UX

This section describes an example of a dApp for Consumers who own >0 LNFTs. dApp should be imagined running on either Desktop, Phone, "Smart" TV, "Smart" Watch.

It does not seek to address Publisher UX, only to present a coherent and simple user interface for a Consumer.

#### Base functionality

1. Consumer launches app, connects wallet

2. dApp shows Consumer's funds to pay tax (like their "viewing credit" amount). Consumer can deposit to / withdraw from these funds, even before owning any LNFTs.

3. dApp shows Consumer's LNFTs (like a "subscribed channels" list).
For each LNFT, the dApp shows
i) the name,
ii) a thumbnail,
iii) the Consumer's sale price (at which another Consumer can force a sale)
iv) the annual tax rate (% rate & absolute amount)
Consumer can edit the sale price for their LNFT at any time

4. If any LNFTs are live, show their live video in the thumbnail, with audio muted. Consumer can unmute audio from the main list, or select to open a view on any of the LNFTs.

For simplicity, this example does not assume any ability to play recordings of previous livestreams (i.e. deadstreams).

### Questions

These are questions that remain, and may prove useful in provoking a discussion:

1. What are the vulnerabilities to the mechanism outlined above?

2. Is the "maximum number of LNFTs that can be minted" necessary, or could this be simplified by making it infinite?

3. Is it sound to allow the Publisher to change i) the mint price, ii) the % tax rate and/or iii) the total supply? The motivation for this choice was to empower the Publisher to be able to retain. Alternatives would be to initially fix:
a) all parameters (seems restrictive, but simplified)
b) some parameters (which one(s)?)

4. If one assumes that the dApp is browser-based, is it possible to enable any kind of "this LNFT has gone live" alerts / notifications. This would necessarily be *without* requiring Broadcastr run any kind of infrastructure. Perhaps using [Push](https://push.org/) or something similar?

5. Can taxes be paid/collected via streaming money in [Superfluid](https://superfluid.finance).

**Comment by wslyvh** [comment]
Not too familiar with Harberger taxes, but from my understanding this is more useful for scarce items. Items that you don't want people to hoard and keep, making it even more scarce and expensive. E.g. physical/unique art, land titles, etc. 
I don't think that's the case for streams/NFTs? Which are essentially abundant ?

It could be interesting to see how popular streams are, if people could scoop up and force buy the NFTs right before a streams. 
Although, that could easily turn into a nasty bidding war where most $$ wins. Not sure that's the outcome a publisher wishes to achieve.

It is interesting though to explore new revenue streams (no pun intended) for creators. Streaming with Suiperfluid for actual pay per view/stream or even pay per minutes streamed would be interesting. E.g. Consumer buy and lock up X amount of $$, but the publisher only receives (as a Superfluid) stream if they hold up to their end and provide new content.
